File: python/src/models/egt_model.py
"""
Basit EGT (egzoz gaz sicakligi) tahmini.
Lineer bir iliski kullanarak yakit akisina dayali hizli hesaplama yapar.
"""

from typing import Union

Number = Union[int, float]


def estimate_egt(fuel_flow: Number, ambient_temp: Number, slope: Number = 2.5) -> float:
    """
    EGT'yi lineer bir modelle tahmin et.

    Parametreler:
        fuel_flow: Yakıt akış miktarı (kg/s veya tutarlı birim).
        ambient_temp: Ortam sıcaklığı (°C).
        slope: Yakıt akışının EGT üzerindeki katsayısı.

    Döndürür:
        Tahmini EGT değeri (°C).
    """
    # Basit model: EGT = ortam sicakligi + slope * yakit akisi
    return float(ambient_temp) + float(slope) * float(fuel_flow)

File: matlab/egt_model.m
% EGT tahmini icin basit lineer model.
% N1 ve yakit akisina dayali hizli hesaplama yapar.
function egt = egt_model(n1, fuel_flow, coeffs)
    % coeffs: [a b c] katsayilari, varsayilan [1.2 2.5 50]
    if nargin < 3 || isempty(coeffs)
        coeffs = [1.2, 2.5, 50];
    end
    a = coeffs(1);
    b = coeffs(2);
    c = coeffs(3);

    % EGT = a*N1 + b*fuel_flow + c
    egt = a .* n1 + b .* fuel_flow + c;
end
